RedfishPkg/RedfishCredentialDxe/RedfishCredentialDxe.h

```diff
@@ -10,68 +11,49 @@
 #ifndef EDKII_REDFISH_CREDENTIAL_DXE_H_
 #define EDKII_REDFISH_CREDENTIAL_DXE_H_
 
-#include <Protocol/EdkIIRedfishCredential.h>
+#include <Protocol/EdkIIRedfishCredential2.h>
 
 #include <Library/BaseLib.h>
 #include <Library/DebugLib.h>
 #include <Library/PrintLib.h>
 #include <Library/RedfishCredentialLib.h>
 #include <Library/UefiLib.h>
 #include <Library/UefiBootServicesTableLib.h>
+#include <Library/RedfishHttpLib.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/RedfishDebugLib.h>
 
-/**
-  Retrieve platform's Redfish authentication information.
-
-  This functions returns the Redfish authentication method together with the user Id and
-  password.
-  - For AuthMethodNone, the UserId and Password could be used for HTTP header authentication
-    as defined by RFC7235.
-  - For AuthMethodRedfishSession, the UserId and Password could be used for Redfish
-    session login as defined by  Redfish API specification (DSP0266).
-
-  Callers are responsible for and freeing the returned string storage.
-
-  @param[in]   This                Pointer to EDKII_REDFISH_CREDENTIAL_PROTOCOL instance.
-  @param[out]  AuthMethod          Type of Redfish authentication method.
-  @param[out]  UserId              The pointer to store the returned UserId string.
-  @param[out]  Password            The pointer to store the returned Password string.
-
-  @retval EFI_SUCCESS              Get the authentication information successfully.
-  @retval EFI_ACCESS_DENIED        SecureBoot is disabled after EndOfDxe.
-  @retval EFI_INVALID_PARAMETER    This or AuthMethod or UserId or Password is NULL.
-  @retval EFI_OUT_OF_RESOURCES     There are not enough memory resources.
-  @retval EFI_UNSUPPORTED          Unsupported authentication method is found.
-
-**/
-EFI_STATUS
-EFIAPI
-RedfishCredentialGetAuthInfo (
-  IN  EDKII_REDFISH_CREDENTIAL_PROTOCOL  *This,
-  OUT EDKII_REDFISH_AUTH_METHOD          *AuthMethod,
-  OUT CHAR8                              **UserId,
-  OUT CHAR8                              **Password
-  );
-
-/**
-  Notify the Redfish service provide to stop provide configuration service to this platform.
-
-  This function should be called when the platfrom is about to leave the safe environment.
-  It will notify the Redfish service provider to abort all logined session, and prohibit
-  further login with original auth info. GetAuthInfo() will return EFI_UNSUPPORTED once this
-  function is returned.
-
-  @param[in]   This                Pointer to EDKII_REDFISH_CREDENTIAL_PROTOCOL instance.
+#define REDFISH_CREDENTIAL_VERSION              0x00010000

```

Comment:
I think the protocol version definition should be placed in EdkIIRedfishCredential2.h together. Below is the example:

https://github.com/tianocore/edk2/blob/a679ceca974e94a659f9b6b9e7a7900644220ef9/MdePkg/Include/Protocol/BlockIo.h#L202-L204

Reply:
I just copied how it was implemented in EDKII_REDFISH_PLATFORM_CONFIG_PROTOCOL protocol.
REDFISH_PLATFORM_CONFIG_VERSION defined in the RedfishPlatformConfigDxe.h
But if you think it should be moved, I will do it)

Reply:
Yes, let's follow how it defined in MdePkg. For Redfish platform config protocol, I need to fix it later.

Reply:
Got it. Will do